pages/[chain]/[address].tsx

```diff
@@ -265,6 +265,12 @@ const SynthPage: React.FC<Props> = ({ data, chainId, relatedSynths }) => {
     toBN("0")
   );
 
+  const showLspForm = useMemo(() => {
+    if (data.type !== "lsp") return false;
+    if (isExpired && !isConnected) return false;
+    return true;
+  }, [data.type, isConnected, isExpired]);
```

Comment:
this is not wrong, but I think we should always prefer flat values in deps vs object access or objects. 
We could destructure `type` before the `useMemo` and use it as a flat value here